File: day01/src/Day01.jl
module Day01

export read_input

function read_input(file = "input.txt")
    elves = Vector{Int}[]
    current_elf = Int[]
    for line in eachline(file)
        if isempty(line)
            push!(elves, current_elf)
            current_elf = Int[]
        else
            push!(current_elf, parse(Int, line))
        end
    end
    push!(elves, current_elf)
    return elves
end

#70369
part1(input = read_input()) = maximum(sum.(input))

#203002
part2(input = read_input()) = sum(sort(sum.(input))[end-2:end])

function solve(file = "input.txt")
    input = read_input(file)
    println(part1(input))
    println(part2(input))
end

__init__() = solve()

end

File: day01/src/day01.py
def read_input(file):
    elves = []
    current_elf = []
    with open(file, "r") as f:
        for line in f:
            line = line[:-1]
            if line:
                current_elf.append(int(line))
            else:
                elves.append(current_elf)
                current_elf = []
        elves.append(current_elf)
    return elves

def part1(input):
    totals = [sum(elf) for elf in input]
    return max(totals)

def part2(input):
    totals = [sum(elf) for elf in input]
    totals.sort()
    return sum(totals[-3:])

input = read_input("input.txt")
print(part1(input))
print(part2(input))
